app/test/service/security_advisory/security_advisory_test.dart

```diff
@@ -1,15 +1,26 @@
-// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
-// for details. All rights reserved. Use of this source code is governed by a
-// BSD-style license that can be found in the LICENSE file.
+import 'dart:io';
 
 import 'package:clock/clock.dart';
+import 'package:path/path.dart' as path;
 import 'package:pub_dev/service/security_advisories/backend.dart';
 import 'package:pub_dev/service/security_advisories/models.dart';
+import 'package:pub_dev/shared/utils.dart';
 import 'package:test/test.dart';
 
 import '../../shared/test_services.dart';
 
 void main() {
+  testWithProfile('Parse advisory', fn: () async {
```

Comment:
Don't test with profile, if you're not really testing services with fake data.

```suggestion
  test('Parse advisory', fn: () async {
```

You could consider making a `app/test/service/security_advisory/osv_test.dart` with a few simple test vectors, I think having the real one from `package:http` is great.

if you want you an also just hardcode in a few fake ones to exercise the code. But this is probably not needed.

Reply:
If you make it a separate `osv_test.dart` file, you won't have many other test cases in the file, so you could just make a big multi-line string with the JSON instead of having a file.